server/auth/auth_test.go

```diff
@@ -1290,21 +1290,34 @@ func TestHandleResponse(t *testing.T) {
 	newID := comms.NextID()
 	client.logReq(newID, func(comms.Link, *msgjson.Message) {},
 		0, func() { t.Log("expired (ok)") })
-	time.Sleep(time.Millisecond) // expire Timer func run in goroutine
-	client.mtx.Lock()
-	if len(client.respHandlers) != 0 {
-		t.Fatalf("expected 0 response handlers, found %d", len(client.respHandlers))
+	// Wait in a loop until response handler expires.
```

Comment:
I think this is often a better approach to test cases like these (compared to just single sleep)

Reply:
I tend to agree.  I think one thing that would encourage taking this extra step if there was a simple helper function:

```go
func waitFor(pred func() bool, timeout time.Duration) (fail bool) {
	tStart := time.Now()
	for {
		if pred() {
			return false
		}
		if time.Since(tStart) > timeout {
			return true
		}
		time.Sleep(10 * time.Millisecond)
	}
}
```

Used in tests like like

```go
	if waitFor(func() bool {
		client.mtx.Lock()
		defer client.mtx.Unlock()
		return len(client.respHandlers) != 0
	}, 10*time.Second) {
		t.Fatal("no response handlers found")
	}
```

Reply:
yeah, I went for helper-func `retryUntilSuccess` in https://github.com/decred/dcrdex/pull/1737 to get rid of code-duplication,

can apply here as well, I guess (done! only seems like predicate should be inverted @chappjc).